### PR TITLE
syncer: support ignore table

### PR DESCRIPTION
**Pull request by nolouch**
Ref https://github.com/pingcap/tidb-tools/issues/50
PTAL @iamxy @qiuyesuifeng  @GregoryIan  @cwen0

### CODE REVIEW

File: syncer/config.go [hunk 1]
```diff
@@ -86,11 +86,16 @@ type Config struct {
...
-	//Ref:http://dev.mysql.com/doc/refman/5.7/en/replication-options-slave.html#option_mysqld_replicate-do-table
+	// Ref: http://dev.mysql.com/doc/refman/5.7/en/replication-options-slave.html#option_mysqld_replicate-do-table
 	DoTable []TableName `toml:"replicate-do-table" json:"replicate-do-table"`
 
 	DoDB []string `toml:"replicate-do-db" json:"replicate-do-db"`
 
+	// Ref: http://dev.mysql.com/doc/refman/5.7/en/replication-options-slave.html#option_mysqld_replicate-ignore-db
+	IgnoreTable []TableName `toml:"replicate-ignore-table" json:"replicate-ignore-table"`
+
+	IgnoreDB []string `toml:"replicate-ignore-db" json:"replicate-ignore-db"`
```
Comment (GregoryIan) [inline syncer/config.go]: s/IgnoreDB/IgnoreDBs/

File: syncer/config.go [hunk 2]
```diff
@@ -86,11 +86,16 @@ type Config struct {
...
 	Meta string `toml:"meta" json:"meta"`
 
-	//Ref:http://dev.mysql.com/doc/refman/5.7/en/replication-options-slave.html#option_mysqld_replicate-do-table
+	// Ref: http://dev.mysql.com/doc/refman/5.7/en/replication-options-slave.html#option_mysqld_replicate-do-table
 	DoTable []TableName `toml:"replicate-do-table" json:"replicate-do-table"`
 
 	DoDB []string `toml:"replicate-do-db" json:"replicate-do-db"`
 
+	// Ref: http://dev.mysql.com/doc/refman/5.7/en/replication-options-slave.html#option_mysqld_replicate-ignore-db
+	IgnoreTable []TableName `toml:"replicate-ignore-table" json:"replicate-ignore-table"`
```
Comment (GregoryIan) [inline syncer/config.go]: s/IgnoreTable/IgnoreTables

File: syncer/config.go [hunk 3]
```diff
@@ -124,6 +129,11 @@ func (c *Config) Parse(arguments []string) error {
 		return errors.Errorf("'%s' is an invalid flag", c.FlagSet.Arg(0))
 	}
 
+	if (len(c.DoDB) != 0 || len(c.DoTable) != 0) &&
+		(len(c.IgnoreDB) != 0 || len(c.IgnoreTable) != 0) {
+		return errors.Errorf("'do' and 'ignore' can not appear at the same time")
```
Comment (GregoryIan) [inline syncer/config.go]: errors.New("xxxxxx")

File: syncer/config.go (hunk 2 above)
Comment (GregoryIan) [inline syncer/config.go]: same with DoDB and DoTable

File: syncer/config.go [hunk 4]
```diff
@@ -134,6 +144,9 @@ func (c *Config) adjust() {
 		c.DoTable[i].Name = strings.ToLower(c.DoTable[i].Name)
 	}
 
+	for i := 0; i < len(c.IgnoreTable); i++ {
+		c.IgnoreTable[i].Name = strings.ToLower(c.IgnoreTable[i].Name)
```
Comment (GregoryIan) [inline syncer/config.go]: what about schema? case sensitive?

File: syncer/syncer.go [hunk 5]
```diff
@@ -534,22 +536,30 @@ func (s *Syncer) skipRowEvent(schema string, table string) bool {
...
 		}
-
 		return true
+	case s.blackFilter:
+		for _, d := range s.cfg.IgnoreTable {
+			if s.matchString(d.Schema, schema) && s.matchString(d.Name, table) {
+				return true
+			}
+		}
+		if s.matchDB(s.cfg.IgnoreDB, schema) && len(s.cfg.IgnoreDB) > 0 {
```
Comment (GregoryIan) [inline syncer/syncer.go]: need len(s.cfg.IgnoreDB) > 0 ?

Comment (GregoryIan) [inline syncer/syncer.go]: the same with `case s.whiteFilter:`

File: syncer/syncer.go [hunk 6]
```diff
@@ -587,26 +597,32 @@ func (s *Syncer) skipQueryDDL(sql string, schema string) bool {
...
-		}
-
-		if tb.Schema == defaultIgnoreDB {
+	if tb.Schema == "" {
+		tb.Schema = schema
+	}
+	if tb.Schema == defaultIgnoreDB {
+		return true
+	}
+	if err == nil {
```
Comment (GregoryIan) [inline syncer/syncer.go]: need check `if err == nil`?

File: syncer/config.go (hunk 4 above)
Comment (nolouch) [inline syncer/config.go]: Mysql is not case sensitive in Windows, but is case sensitive in most varieties of Unix. and TiDB now is not case sensitive. any suggest?

File: syncer/syncer.go (hunk 6 above)
Comment (nolouch) [inline syncer/syncer.go]: we need to check it, it represents the sql is a ddl sql.

File: syncer/README.md [hunk 7]
```diff
@@ -53,6 +53,31 @@ host = "127.0.0.1"
...
+[[replicate-do-table]]
+db-name ="db3"
+tbl-name = "table2"
+
+# not synchronize table3 under db1
+[[replicate-ignore-table]]
+db-name = "db1"
+tbl-name = "table3"
+
+# supports regular, beginning with ~
```
Comment (andelf) [inline syncer/README.md]: s/regular/regexp/

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**